utilities/xmlrpcpp/src/XmlRpcSocket.cpp

```diff
@@ -317,6 +317,13 @@ XmlRpcSocket::nbRead(int fd, std::string& s, bool *eof)
       return false;   // Error
     }
   }
+  // Watch for integer overrun
+  if (s.length() > size_t(__INT_MAX__)) {
+    XmlRpcUtil::error("XmlRpcSocket::nbRead: text size (%u) exceeds the maximum allowed size (%s).",
+                      s.length(), __INT_MAX__);
+    s.resize(__INT_MAX__);
```

Comment:
I don't understand this truncation; what will happen to the data in `s` after this? Will it be garbage-collected? Or passed on to something else for further handling? I suspect discarding all the data would be easier / better than keeping two gigabytes of the data.

Reply:
Changed to clear() in 7d51a7aaea0cf022cc82357e03db6340741364d3